crates/bevy_ecs/src/system/query.rs

```diff
@@ -275,6 +279,41 @@ where
         }
     }
 
+    /// Creates a new [`Query`] from a [`QueryState`]
```

Comment:
It is important to call out that Query will not update QueryState to account for changes in World. If you create a QueryState, do something that creates a new archetype, then construct a Query from it, the new archetype will be ignored.

Using Query methods is essentially equivalent to the `QueryState::X_manual` methods.